test/integration/sdf_dom_conversion.cc

```diff
@@ -402,3 +404,163 @@ TEST(SDFDomConversion, Sensors)
+
+  // Get the first link
+  const sdf::Link *link = model->LinkByIndex(0);
+  ASSERT_NE(nullptr, link);
+
+  // spot
+  {
+    const sdf::Light *light = link->LightByName("link_spot_light");
+    sdf::ElementPtr lightElem = light->ToElement();
+    auto spotLight = std::make_unique<sdf::Light>();
+    spotLight->Load(lightElem);
+    ASSERT_NE(nullptr, spotLight);
+    EXPECT_EQ("link_spot_light", spotLight->Name());
+    EXPECT_EQ(sdf::LightType::SPOT, spotLight->Type());
+    EXPECT_EQ(ignition::math::Pose3d::Zero,
+        spotLight->RawPose());
+    EXPECT_TRUE(spotLight->CastShadows());
+    EXPECT_DOUBLE_EQ(12.35, spotLight->AttenuationRange());
+    EXPECT_DOUBLE_EQ(0.1, spotLight->LinearAttenuationFactor());
+    EXPECT_DOUBLE_EQ(0.2, spotLight->ConstantAttenuationFactor());
+    EXPECT_DOUBLE_EQ(0.001, spotLight->QuadraticAttenuationFactor());
+    EXPECT_EQ(ignition::math::Color(0.0f, 0.05f, 0.06f, 1.0f),
+        spotLight->Diffuse());
+    EXPECT_EQ(ignition::math::Color(0.03f, 0.02f, 0.0f, 1.0f),
+        spotLight->Specular());
+    EXPECT_EQ(ignition::math::Vector3d(1.0, 5.0, 2.0), spotLight->Direction());
+  }
+
+  // point
+  {
+    const sdf::Light *light = link->LightByName("link_point_light");
+    sdf::ElementPtr lightElem = light->ToElement();
+    auto pointLight = std::make_unique<sdf::Light>();
+    pointLight->Load(lightElem);
+    ASSERT_NE(nullptr, pointLight);
+    EXPECT_EQ("link_point_light", pointLight->Name());
+    EXPECT_EQ(sdf::LightType::POINT, pointLight->Type());
+    EXPECT_EQ(ignition::math::Pose3d(10, 20, 100, 0, 0, 0),
+        pointLight->RawPose());
+    EXPECT_FALSE(pointLight->CastShadows());
+    EXPECT_EQ(ignition::math::Color(1.0f, 0.0f, 0.6f, 1.0f),
+        pointLight->Diffuse());
+    EXPECT_EQ(ignition::math::Color(0.3f, 0.2f, 0.0f, 1.0f),
+        pointLight->Specular());
+    EXPECT_DOUBLE_EQ(1235.0, pointLight->AttenuationRange());
+    EXPECT_DOUBLE_EQ(1.0, pointLight->LinearAttenuationFactor());
+    // negative value should be clamped to 0
+    EXPECT_DOUBLE_EQ(0.0, pointLight->ConstantAttenuationFactor());
+    EXPECT_DOUBLE_EQ(10.2, pointLight->QuadraticAttenuationFactor());
+  }
+
+  // directional
+  {
+    const sdf::Light *light = link->LightByName("link_directional_light");
+    sdf::ElementPtr lightElem = light->ToElement();
+    auto dirLight = std::make_unique<sdf::Light>();
+    dirLight->Load(lightElem);
+    ASSERT_NE(nullptr, dirLight);
+    EXPECT_EQ("link_directional_light", dirLight->Name());
+    EXPECT_EQ(sdf::LightType::DIRECTIONAL, dirLight->Type());
+    EXPECT_EQ(ignition::math::Pose3d(0, 1, 2, 0, 0, 0),
+        dirLight->RawPose());
+    EXPECT_TRUE(dirLight->CastShadows());
+    EXPECT_EQ(ignition::math::Color(0.0f, 1.0f, 0.2f, 1.0f),
+        dirLight->Diffuse());
+    EXPECT_EQ(ignition::math::Color(0.0f, 0.2f, 0.1f, 1.0f),
+        dirLight->Specular());
+    EXPECT_EQ(ignition::math::Vector3d(4.0, 5.0, 6.0), dirLight->Direction());
+  }
```

Comment:
I think the tests are good, but they don't cover the use case where an `sdf::Light` is created by the user *without* calling `sdf::Light::Load`. But the main advantage of `ToElement` is to be used in that scenario or when a DOM object is modified after `Load` is called. Otherwise, we would just use the existing `Element()` function.

Reply:
something like a9fedbd64ccdcfaf26a54d449092c84e1501e816? The new test checks the values produced by `ToElement` without calling `Load` and also checks the case when values are modified then `ToElement` is called.